spyder_line_profiler/spyder/widgets.py

```diff
@@ -100,123 +112,217 @@ def natural_sort_key(key):
+        layout = QVBoxLayout()
+        layout.addWidget(self.datatree)
+        self.setLayout(layout)
+        
+        # Signals
+        self.datatree.sig_edit_goto_requested.connect(
+            self.sig_edit_goto_requested)
+        
+    # --- PluginMainWidget API
+    # ------------------------------------------------------------------------
+    
+    def create_stretcher(self, id_):
+        # for backwards compatibility to Spyder<5.1.5, where no id_ is used
+        try:
+            return PluginMainWidget.create_stretcher(self, id_)
+        except Exception:
+            return PluginMainWidget.create_stretcher(self)
+    
+    def get_title(self):
+        return _("Line Profiler")
+
+    def get_focus_widget(self):
+        pass
+
+    def setup(self):
+        
+        self.start_action = self.create_action(
+            SpyderLineProfilerWidgetActions.Run,
             text=_("Profile by line"),
             tip=_("Run line profiler"),
-            triggered=(lambda checked=False: self.analyze()), text_beside_icon=True)
-        self.stop_button = create_toolbutton(
-            self,
-            icon=get_icon('terminate.png'),
+            icon=self.create_icon('run'),
+            triggered=self.start,
+        )
+        self.stop_action = self.create_action(
+            SpyderLineProfilerWidgetActions.Stop,
             text=_("Stop"),
             tip=_("Stop current profiling"),
-            text_beside_icon=True)
-        self.filecombo.valid.connect(self.start_button.setEnabled)
-        #self.filecombo.valid.connect(self.show_data)
-        # FIXME: The combobox emits this signal on almost any event
-        #        triggering show_data() too early, too often.
-
-        browse_button = create_toolbutton(
-            self, icon=get_icon('fileopen.png'),
-            tip=_('Select Python script'),
-            triggered=self.select_file)
-
-        self.datelabel = QLabel()
-
-        self.log_button = create_toolbutton(
-            self, icon=get_icon('log.png'),
-            text=_("Output"),
-            text_beside_icon=True,
+            icon=self.create_icon('stop'),
+            triggered=self.kill_if_running,
+        )
+        self.browse_action = self.create_action(
+            SpyderLineProfilerWidgetActions.Browse,
+            text=_("Open Script"),
+            tip=_('Select Python script'),            
+            icon=self.create_icon('fileopen'),
+            triggered=self.select_file,
+        )
+        self.log_action = self.create_action(
+            SpyderLineProfilerWidgetActions.ShowOutput,
+            text=_("Show Result"),
             tip=_("Show program's output"),
-            triggered=self.show_log)
-
-        self.datatree = LineProfilerDataTree(self)
-
-        self.collapse_button = create_toolbutton(
-            self,
-            icon=get_icon('collapse.png'),
+            icon=qta.icon("fa.file-text", color=ima.MAIN_FG_COLOR),
```

Comment:
I checked locally and this should be:

```suggestion
            icon=self.create_icon('log'),
```
Which will make it look like this:
![imagen](https://user-images.githubusercontent.com/16781833/159551013-ea3f1d12-4cbf-4e88-b2c7-9c9f9989d93c.png)